airbyte-integrations/connectors/source-chargebee/source_chargebee/source.py

```diff
@@ -22,205 +22,51 @@
-        if latest_updated_at:
-            current_stream_state.update(
-                {
-                    "update_at[after]": latest_updated_at,
-                }
-            )
-
-        return current_stream_state
-
-
-class SubscriptionStream(ChargebeeStream):
-    name = "subscription"
-    api = Subscription
 
-
-class CustomerStream(ChargebeeStream):
-    name = "customer"
-    api = Customer
-
-
-class InvoiceStream(ChargebeeStream):
-    name = "invoice"
-    api = Invoice
-
-
-class OrderStream(ChargebeeStream):
-    name = "order"
-    api = Order
-
-
-class PlanStream(ChargebeeStream):
-    name = "plan"
-    api = Plan
-
-
-class AddonStream(ChargebeeStream):
-    name = "addon"
-    api = Addon
+from .streams import Addon, Customer, Invoice, Order, Plan, Subscription
 
 
 class SourceChargebee(AbstractSource):
-    # Class variables
-    LIMIT = 100
-
     def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
         # Configure the Chargebee Python SDK
-        chargebee.configure(
-            api_key=config["site_api_token"],
-            site=config["site"],
-        )
+        chargebee.configure(api_key=config["site_api_key"], site=config["site"])
         try:
-            # Get one subscription to test connection
-            Subscription.list(
-                # Set limit
-                # to test on a small dataset
-                params={
-                    "limit": 1,
-                },
-            )
+            subscription_stream = Subscription(start_date=config["start_date"])
+            next(subscription_stream.read_records(sync_mode=SyncMode.full_refresh))
             return True, None
         except Exception as err:
-            # Should catch all exceptions
-            # which are already handled by
-            # Chargebee Python wrapper
+            # Should catch all exceptions which are already handled by Chargebee Python wrapper.
             # https://github.com/chargebee/chargebee-python/blob/5346d833781de78a9eedbf9d12502f52c617c2d2/chargebee/http_request.py
-            return False, str(err)
+            return False, repr(err)
 
     def streams(self, config) -> List[Stream]:
         # Configure the Chargebee Python SDK
-        chargebee.configure(
-            api_key=config["site_api_token"],
-            site=config["site"],
-        )
-        # Add the streams
-        streams = [
-            SubscriptionStream(),
-            CustomerStream(),
-            InvoiceStream(),
-            OrderStream(),
-            PlanStream(),
-            AddonStream(),
+        chargebee.configure(api_key=config["site_api_key"], site=config["site"])
+
+        kwargs = {"start_date": config["start_date"]}
+        product_catalog_version = config["product_catalog"]
+
+        # Below streams are suitable for both `Product Catalog 1.0` and `Product Catalog 2.0`.
+        common_streams = [
+            Customer(**kwargs),
```

Comment:
is it hard to also add the item streams? (the v2 replacement of plan)

Reply:
Nope, it's not hard. I've created a follow-up ticket for adding 3 additional streams: `Itesm`, `Item prices` and `Attached Items` and I was planning to add them right after this PR is merged and the first connector version is pubished.